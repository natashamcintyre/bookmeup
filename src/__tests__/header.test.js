import React from 'react'
import Header from '../components/header.js'

describe('header', () => {
  it('renders without crashing', () => {
    const component = mount(<Header />)
    expect(component).toMatchSnapshot()
  })

  it('has a logo', () => {
    const component = mount(<Header />)
    expect(component.exists('img')).toBe(true)
  })

  it('shows modal', () => {
    const component = mount(<Header />)
    component.find('button#isbnSearchButton').simulate('click')
    expect(component.find('div#isbnSearchModal').hasClass('modal display-block')).toBe(true);
  })

  it('hides modal', () => {
    const component = mount(<Header />)
    component.find('button#isbnSearchButton').simulate('click')
    component.find('#closeIsbnSearch').simulate('click')
    expect(component.find('div#isbnSearchModal').hasClass('modal display-none')).toBe(true);
  })
})
