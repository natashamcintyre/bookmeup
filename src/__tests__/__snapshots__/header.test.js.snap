// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header renders without crashing 1`] = `ReactWrapper {}`;
